x-hooks/use-octokit.tsx

```diff
@@ -0,0 +1,101 @@
+import { Octokit } from "octokit";
+import { useState } from "react";
+
+const mapData = ({data}) => data;
+
+const GITHUB_STATS_TTL = 24 * 60 * 60 * 1000
+
+const githubRepoStats = {
+  lastUpdated: 0,
+  data: {},
+}
+
+const githubForkStats = {
+  lastUpdated: 0,
+  data: {},
+}
+
+export default function useOctokit() {
+  const [octokit, setOctokit] = useState<Octokit>(new Octokit({auth: process.env.NEXT_PUBLIC_GITHUB_TOKEN}));
```

Comment:
we will not be using a bot token on this client, so we can't provide auth - as it would defeat the whole porpose of having the user make this connection